### ISSUE TITLE
[BUG] Unable to extract template

### ISSUE THREAD

**Issue by badsyntax** [OpenAPITools/openapi-generator#9557]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I'm following the guidelines here: https://openapi-generator.tech/docs/usage/

I'm unable to extract template.

I run:

```console
openapi-generator-cli author template -g java
```

 I get the following error:

```console
Did set selected version to 5.1.1
[error] Found unexpected parameters: [-g, java]

See 'openapi-generator-cli help' for usage.
```

##### openapi-generator version

`5.1.1.`

##### OpenAPI declaration file content or url

N/A

##### Generation Details

n/a

##### Steps to reproduce

```console
npm i -g @openapitools/openapi-generator-cli
openapi-generator-cli author template -g java
```

**Comment by wing328** [comment]
Works for me with the latest stable version v5.2.1
```
williamcheng ~/Code/openapi-generator (master) $ openapi-generator-cli version
5.2.1
williamcheng ~/Code/openapi-generator (master) $ openapi-generator-cli author template -g java
[main] INFO  o.o.codegen.cmd.AuthorTemplate - Extracted templates to 'out' directory. Refer to https://openapi-generator.tech/docs/templating for customization details.
```

Can you use the latest version and give it another try?

**Comment by badsyntax** [comment]
It's so odd, i'm still getting the error:

```console
❯ openapi-generator-cli version
5.2.1

❯ openapi-generator-cli author template -g java
[error] Found unexpected parameters: [-g, java]

See 'openapi-generator-cli help' for usage.
```

I've tried node 14.17.3 & 16.7.0, bash and zsh. I'm on an Apple M1.

**Comment by badsyntax** [comment]
@wing328 I don't understand why this is working for you. I cloned the cli project and ran locally and was still able to reproduce. I've added a PR to fix: https://github.com/OpenAPITools/openapi-generator-cli/pull/448

**Comment by badsyntax** [comment]
@wing328 This is a cli issue not a generator issue. I'm confident you'll be able to replicate if you upgrade your global openapi-generator-cli package to the latest version.

**Comment by syndicatedshannon** [comment]
I did experience it yesterday with the latest CLI:
NPM: @openapitools/openapi-generator-cli@2.3.10
JAR: 5.2.1

That was why I noted it over here:
https://github.com/OpenAPITools/openapi-generator-cli/issues/368#issuecomment-903346179

**Comment by kay-schecker** [comment]
fixed in @openapitools/openapi-generator-cli@2.4.0

**Comment by badsyntax** [comment]
Verified fixed with https://github.com/OpenAPITools/openapi-generator-cli/releases/tag/v2.4.0
